src/celeritas/ext/detail/GeantSurfacePhysicsLoader.cc

```diff
@@ -0,0 +1,427 @@
+
+            GSPL_OSF_PAIR(groundlumirrorair),
+            GSPL_OSF_PAIR(groundlumirrorglue),
+            GSPL_OSF_PAIR(groundair),
+            GSPL_OSF_PAIR(groundteflonair),
+            GSPL_OSF_PAIR(groundtioair),
+            GSPL_OSF_PAIR(groundtyvekair),
+            GSPL_OSF_PAIR(groundvm2000air),
+            GSPL_OSF_PAIR(groundvm2000glue),
+
+            GSPL_OSF_PAIR(Rough_LUT),
+            GSPL_OSF_PAIR(RoughTeflon_LUT),
+            GSPL_OSF_PAIR(RoughESR_LUT),
+            GSPL_OSF_PAIR(RoughESRGrease_LUT),
+
+            GSPL_OSF_PAIR(Polished_LUT),
+            GSPL_OSF_PAIR(PolishedTeflon_LUT),
+            GSPL_OSF_PAIR(PolishedESR_LUT),
+            GSPL_OSF_PAIR(PolishedESRGrease_LUT),
+
+            GSPL_OSF_PAIR(Detector_LUT),
+        };
+
+    if (auto iter = names.find(value); iter != names.end())
+    {
+        return names.find(value)->second;
+    }
+    return "UNKNOWN";
+
+#undef GSPL_OSF_PAIR
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Verify that all elements of a grid are within range [0, 1].
+ *
+ * Used to verify that \c ReflectionForm Grids are within the expected range.
+ */
+bool unity(inp::Grid const& grid)
+{
+    return std::any_of(grid.y.begin(), grid.y.end(), [](real_type const& val) {
+        return val >= 0 && val <= 1;
+    });
+}
+
+//---------------------------------------------------------------------------//
+}  // namespace
+
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with \c SurfacePhysics input to be filled by \c operator() .
+ */
+GeantSurfacePhysicsLoader::GeantSurfacePhysicsLoader(inp::SurfacePhysics& result)
+    : result_(result)
+{
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Populate surface physics data for a given \c SurfaceId .
+ */
+void GeantSurfacePhysicsLoader::operator()(SurfaceId sid)
+{
+    CELER_EXPECT(sid);
+    detail::GeantSurfacePhysicsHelper helper(sid);
+    try
+    {
+        this->insert_reflectivity(helper);
+        this->insert_roughness(helper);
+        this->insert_interaction(helper);
+        this->validate_model(helper);  // Verify model requirements
+    }
+    catch (RuntimeError const& e)
+    {
+        throw;
+    }
+
+    CELER_LOG(debug) << "Inserted surface id " << sid.unchecked_get()
+                     << " with " << to_cstring(helper.surface().GetModel())
+                     << " model and "
+                     << to_cstring(helper.surface().GetFinish()) << " finish";
+}
+
+//---------------------------------------------------------------------------//
+// PRIVATE MEMBER FUNCTIONS
+//---------------------------------------------------------------------------//
+/*!
+ * Collect reflectivity information from a given optical surface.
+ */
+void GeantSurfacePhysicsLoader::insert_reflectivity(
+    detail::GeantSurfacePhysicsHelper& helper)
```

Comment:
Since the surface physics loader is in the detail namespace, you don't have to add `detail::` here.